lib/plaindate.ts

```diff
@@ -208,21 +208,30 @@ export class PlainDate implements Temporal.PlainDate {
   toZonedDateTime(item: Params['toZonedDateTime'][0]): Return['toZonedDateTime'] {
     if (!ES.IsTemporalDate(this)) throw new TypeError('invalid receiver');
 
-    type TimeZoneAndPlainTimeProps = Exclude<typeof item, Temporal.TimeZoneProtocol | string>;
-    let timeZone: Temporal.TimeZoneProtocol, temporalTime: TimeZoneAndPlainTimeProps['plainTime'];
+    type TimeZoneAndPlainTimeProps = Exclude<typeof item, string | Temporal.TimeZoneProtocol>;
+    let timeZone: Temporal.TimeZoneLike, temporalTime: TimeZoneAndPlainTimeProps['plainTime'];
     if (ES.IsObject(item)) {
       if (ES.IsTemporalTimeZone(item)) {
         timeZone = item;
       } else {
-        if (item.timeZone === undefined) {
-          timeZone = ES.ToTemporalTimeZone(item);
+        const timeZoneLike = (item as TimeZoneAndPlainTimeProps).timeZone;
+        if ((item as TimeZoneAndPlainTimeProps).timeZone === undefined) {
```

Comment:
Yep, I just saw this too. I suspect I originally put it there to avoid the TS cast, but the property access is observable (for a Proxy) so there's no reason to do it twice. Also it differs from upstream which only accesses the property once.

I'll fix at the same time as your comment below.